### ISSUE TITLE
Automated testing framework

### ISSUE THREAD

**Issue by dvalters** [LSDtopotools/LSDTopoTools2_roadmap#3]
**Regression testing**:
We could gently introduce some sort of basic regression testing for each major type of analysis. i.e. some standard DEM input and output

**Unit tests**? 
Not suggesting we go back and write unit test for every single function, but could try out something like googletest or other C++ unit testing frameworks.

**Comment by dvalters** [comment]
This could all be hooked up to some CI server... :heart_eyes:

**Comment by sgrieve** [comment]
I agree, getting everything unit tested would require an insane amount of work. But regression testing the key components would be amazing.

My preference for a framework is [catch](https://github.com/catchorg/Catch2) but happy to look at other options.

We can either run things on travis alongside nightly builds of docs, or we can look at testing more compute intensive stuff using the new uk [Jenkins instance] (http://www.softeng-support.ac.uk/buildservice/)

**Comment by sgrieve** [comment]
An option for getting unit tests developed would be to try and host Google summer of code students next year, if we can get the project into a decent state.
